src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1810,9 +1810,23 @@ internal Symbol ContainingMember()
             return containingMember;
         }
 
-        private BoundExpression TryBindInteractiveReceiver(SyntaxNode syntax, Symbol currentMember, NamedTypeSymbol currentType, NamedTypeSymbol memberDeclaringType)
+        private bool IsInstanceContext()
         {
-            if (currentType.TypeKind == TypeKind.Submission && currentMember.RequiresInstanceReceiver())
+            var containingMember = this.ContainingMemberOrLambda;
+            while (containingMember.Kind != SymbolKind.NamedType && (object)containingMember.ContainingSymbol != null && containingMember.ContainingSymbol.Kind != SymbolKind.NamedType)
```

Comment:
> && (object)containingMember.ContainingSymbol != null && containingMember.ContainingSymbol.Kind != SymbolKind.NamedType [](start = 64, length = 119)

Can we avoid checking `Kind != SymbolKind.NamedType` and `IsStatic` twice, without making the loop unnecessarily complicated?

Would something like the following work?
```C#
    var containingMember = this.ContainingMemberOrLambda;
    do
    {
        if (containingMember.IsStatic)
        {
            return false;
        }
        if (containingMember.Kind == SymbolKind.NamedType)
        {
            break;
        }
        containingMember = containingMember.ContainingSymbol;
    } while ((object)containingMember != null);
    return true;

```